src/main/java/org/jenkinsci/plugins/github/webhook/WebhookManager.java

```diff
@@ -141,8 +141,23 @@ public void run() {
      */
     public void unregisterFor(GitHubRepositoryName name, List<GitHubRepositoryName> aliveRepos) {
         try {
-            GHRepository repo = checkNotNull(
-                    from(name.resolve(allowedToManageHooks())).firstMatch(withAdminAccess()).orNull(),
+            /* This is overcomplicated to satisfy unit tests that expect certain codepaths to be called */
+            Iterable<GHRepository> manageableIterable = name.resolve(allowedToManageHooks());
+            GHRepository repo = from(manageableIterable).firstMatch(withAdminAccess()).orNull();
+
+            if (!(manageableIterable.iterator().hasNext())) {
+                if (repo == null) {
```

Comment:
repo will be always null if there is no repos in iterator

Reply:
Hm, makes sense :)